src/cart/utils/memcheck-cart.supp

```diff
@@ -35,6 +35,120 @@
    ...
    fun:sock_pe_progress_thread
 }
+{
+    libopen-pal
+    Memcheck:Leak
+    ...
+    obj:*/libopen-pal.so.40.10.3
+}
+{
+    ld-2
+    Memcheck:Leak
+    ...
+    obj:*/ld-2.17.so
+}
+{
+    libdl-2
+    Memcheck:Leak
+    ...
+    obj:*/libdl-2.17.so
+}
```

Comment:
number of those new suppressions look way too generic. for example libfabric supression on line 56-61 seems to cover everything possible that might leak in it.
For those we should be less generic, suppressing only current issue (along possibly with making a note to submit new issue with ofi team)

Reply:
Removing the generic suppressions leaves 2300 leaks (80 unique functions across 4 unique .so objects).  I've added 18 new specific suppressions, and will try to whittle down the number.  What's a good way to systematically categorize these, and group them into suppression blocks?  It would be handy if we had some sort of tool to do unsupervised text classification on these stack traces.